### ISSUE TITLE
Terraform not support stockholm(eu-north-1) region

### ISSUE THREAD

**Issue by selvakann** [hashicorp/terraform#21319]
provider "aws"{
region="eu-north-1"

}

terraform {
  backend "s3" {
    bucket = "terraform-state1"
    key    = "terraform.tfstate"
    region = "eu-north-1"
  }
}

the above code giving below issue.

Error initializing new backend:
Error configuring the backend "s3": Not a valid region: eu-north-1

Please update the configuration in your Terraform files to fix this error
then run this command again.

regards
Kannan.R

**Comment by apparentlymart** [comment]
Hi @selvakann,

The AWS provider and the S3 backend both live in different codebases where and so their support for regions changes separately. While the necessary code _is_ shared between the two, it is embedded in each program separately and so updating the AWS provider to a version with `eu-north-1` support does not automatically make the S3 backend support that region.

With that said, I'm not sure if this new region is already supported in the AWS provider. If it is, then we can fix this by updating the library we share with the AWS provider that handles interpretation of these settings.

If it is already supported in the AWS provider and it's just the S3 backend that is lagging, you may be able to work around this using the `skip_region_validation` argument:

```hcl
terraform {
  backend "s3" {
    bucket = "terraform-state1"
    key    = "terraform.tfstate"
    region = "eu-north-1"

    # remove this setting once eu-north-1 is supported in the S3 backend
    skip_region_validation = true
  }
}
```

That argument is marked as deprecated in latest versions, but I believe it should still work in all stable versions of Terraform available as I'm writing this.

The AWS provider team usually drives adding support for new regions, so if the AWS provider _itself_ doesn't have support for this region yet then that work will need to happen first, and then we can update the S3 backend to match it.

Sorry for the friction here!

**Comment by dagjsolheim** [comment]
Hi
Are using Terraform v0.11.13
+ provider.aws v2.11.0
+ provider.template v2.1.2

It dos not help using skip_region_validation = true or skip_region_validation = "true"
 The problem been around for a while and make a lot frustration.

**Comment by apparentlymart** [comment]
Hi again @selvakann, and hi @dagjsolheim,

When I wrote my previous reply, Terraform 0.11.13 was the current version. In the meantime, we released both Terraform 0.11.14 and Terraform 0.12.0, both of which should have support for the new AWS regions `eu-north-1` and `us-gov-east-1` in their S3 backends.
